cfs_config_util/parser.py

```diff
@@ -44,6 +45,66 @@
 SAVE_TO_CFS_OPTION = '--save-to-cfs'
 SAVE_SUFFIX_OPTION = '--save-suffix'
 
+ASSIGN_TO_XNAMES_OPTION = '--assign-to-xnames'
+ASSIGN_TO_QUERY_OPTION = '--assign-to-query'
+
+CLEAR_STATE_OPTION = '--clear-state'
+CLEAR_ERROR_OPTION = '--clear-error'
+ENABLE_OPTION = '--enable'
+DISABLE_OPTION = '--disable'
+
+DESIRED_CONFIG_OPTION = '--desired-config'
+
+CANONICAL_UPDATE_CONFIGS_ACTION = 'update-configs'
+CANONICAL_UPDATE_COMPONENTS_ACTION = 'update-components'
+
+
+def convert_comma_separated_list(comma_separated_str):
+    """Convert a comma-separated list into a list.
+
+    Args:
+        comma_separated_str (str): the comma-separated list string to convert
+
+    Returns:
+        list: the converted list
+    """
+    return comma_separated_str.split(',')
+
+
+def convert_query_to_dict(query_str):
+    """Convert a comma-separated list of query parameters to a dictionary.
+
+    This splits a list of comma-separated items of the form "param=value" and
+    adds them to a dictionary mapping from param to the list of values for that
+    param. Multiple instances of a single param are added to a list of values.
+
+    For example "role=management,subrole=master,subrole=storage" becomes:
+
+    {
+        "role": ["management"]
+        "subrole": ["master", "storage"]
+    }
+
+    Args:
+        query_str (str): the comma-separated parameters
+
+    Returns:
+        dict: a dictionary mapping from parameter names to their string value
+            or values.
+    """
+    params = defaultdict(list)
+    for query in query_str.split(','):
+        try:
+            param, value = query.split('=', maxsplit=1)
+        except ValueError:
+            raise argparse.ArgumentTypeError(
+                f'Invalid query string "{query_str}". Query string must consist '
+                f'of one or more comma-separated key=value pairs.'
+            )
```

Comment:
raise from err

Reply:
With this one, I think it's less necessary to keep that history, since this exception is handled by `argparse`, and the `ValueError` is just an implementation detail about how we're checking for the condition reported in the more readable string I've written in this `ArgumentTypeError`.